homeassistant/components/github/sensor.py

```diff
@@ -132,7 +132,7 @@ def update(self):
         self._github_data.update()
 
         self._name = self._github_data.name
-        self._state = self._github_data.latest_commit_sha
+        self._state = self._github_data.latest_release_url
```

Comment:
Try this:
```py
self._state = self._github_data.latest_release_url.split("tag/")[1]
```

Reply:
yes that would be fine, but some repos don't have a tag set, hence I made the template sensor to show Not set in that case.
What would show if the self._state = self._github_data.latest_release_url.split("tag/")[1] would be used? An empty line would bu suboptimal imho.

Reply:
Set it to `None` in that case which will translate to unknown.

Reply:
Isn't it already set to None be default? https://github.com/home-assistant/home-assistant/blob/23f65a7aca3495750e82af20a3a32596acf828d2/homeassistant/components/github/sensor.py#L76

Reply:
If we want to avoid setting it to an empty string we need to check for that case and at least not update it in that case.

Reply:
could this be done:

```
    @property
    def state(self):
        """Return the state of the sensor."""
        if self._latest_release_url is not None:
            return self._state
        else:
            return "Not set"
```

Reply:
We should use `None` to represent unknown state.

Reply:
``` 
   @property
    def state(self):
        """Return the state of the sensor."""
        if self._latest_release_url is not None:
            return self._state
        else:
            return None
```

Reply:
I would move the checks to the update method and set `self._state` appropriately. Then always return `self._state` here in `state`.

Reply:
Ok I understand, that is much better.
In jinja I would do that as below, but I ammunsure what the correct python syntax would be for line [135](https://github.com/home-assistant/home-assistant/blob/23f65a7aca3495750e82af20a3a32596acf828d2/homeassistant/components/github/sensor.py#L135)

```
self._state = self._github_data.latest_release_url.split("tag/")[1] 
                         if self._latest_release_url is not None else None
```

Reply:
That works, but we prefer to write a regular if else block on multiple lines if the ternary operator, in your example, spans more than one line.

```py
if self._latest_release_url is not None:
    self._state = self._github_data.latest_release_url.split("tag/")[1]
else:
    self._state = None
```

Reply:
a ok! I tried to keep the left side as, and only change the right hand side of the update method. will adapt now with your code. Thanks very much for educating me!

omt: does this also replace the line 135 in the update method? If not, we would have 2 data collectors for self_state....?

Reply:
That looks ok.

Side note: It's easier to just make the change and push to github so I can comment on the changed code. Then we don't have to copy and write code examples.